File: beginner/javascript/1049.js
var input = require('fs').readFileSync('stdin', 'utf8');

var valores = input.split('\n');

var p1 = valores.shift();
var p2 = valores.shift();
var p3 = valores.shift();

if (p1 === "vertebrado") {
    if (p2 === "ave") {
        if (p3 === "carnivoro") {
            console.log("aguia");
        } else if (p3 === "onivoro") {
            console.log("pomba");
        }
    } else if (p2 === "mamifero") {
        if (p3 === "onivoro") {
            console.log("homem");
        } else if (p3 === "herbivoro") {
            console.log("vaca");
        }
    }
} else if (p1 === "invertebrado") {
    if (p2 === "inseto") {
        if (p3 === "hematofago") {
            console.log("pulga");
        } else if (p3 === "herbivoro") {
            console.log("lagarta");
        }
    } else if (p2 === "anelideo") {
        if (p3 === "hematofago") {
            console.log("sanguessuga");
        } else if (p3 === "onivoro") {
            console.log("minhoca");
        }
    }
}

File: beginner/c/1049.c
#include <stdio.h>
#include <string.h>

int main()
{
    char p1[50], p2[50], p3[50];

    scanf("%s", p1);
    scanf("%s", p2);
    scanf("%s", p3);

    if (strcmp(p1, "vertebrado") == 0) {
        if (strcmp(p2, "ave") == 0) {
            if (strcmp(p3, "carnivoro") == 0) {
                printf("aguia\n");
            } else if (strcmp(p3, "onivoro") == 0) {
                printf("pomba\n");
            }
        } else if (strcmp(p2, "mamifero") == 0) {
            if (strcmp(p3, "onivoro") == 0) {
                printf("homem\n");
            } else if (strcmp(p3, "herbivoro") == 0) {
                printf("vaca\n");
            }
        }

    } else if (strcmp(p1, "invertebrado") == 0) {
        if (strcmp(p2, "inseto") == 0) {
            if (strcmp(p3, "hematofago") == 0) {
                printf("pulga\n");
            } else if (strcmp(p3, "herbivoro") == 0) {
                printf("lagarta\n");
            }
        } else if (strcmp(p2, "anelideo") == 0) {
            if (strcmp(p3, "hematofago") == 0) {
                printf("sanguessuga\n");
            } else if (strcmp(p3, "onivoro") == 0) {
                printf("minhoca\n");
            }
        }
    }

    return 0;
}
